vllm/config.py

```diff
@@ -179,7 +180,7 @@ def __init__(self,
                                     code_revision, rope_scaling, rope_theta,
                                     config_format)
         self.hf_text_config = get_hf_text_config(self.hf_config)
-        self.pooling_config = self.get_pooling_config()
+        self.pooling_config = pooling_config or self.get_pooling_config()
```

Comment:
Since the two new flags can be used independently, I think we need to be more careful here. The flags are used to override whatever is loaded from the config files. By putting both options in the pooling_config, we risk overriding `normalize` if only `pooling-type` is set and vice-versa.